modules/packages/BufferedAtomics.chpl

```diff
@@ -17,6 +17,63 @@
  * limitations under the License.
  */
 
+/*
+   This module provides buffered versions of non-fetching atomic operations for
+   all ``int``, ``uint``, and ``real`` types.  Buffered versions of
+   :proc:`~Atomics.add()`, :proc:`~Atomics.sub()`, :proc:`~Atomics.or()`,
+   :proc:`~Atomics.and()`, and :proc:`~Atomics.xor()` are provided. These
+   variants are internally buffered and the buffers are flushed implicitly when
+   full or explicitly with :proc:`flushAtomicBuff()`. These buffered operations
+   can provide a significant speedup for bulk atomic operations that do not
+   require strict ordering of operations:
+
+   .. code-block:: chapel
+
+     use BufferedAtomics;
+
+     const numTasksPerLocale = here.maxTaskPar,
+           iters = 10000;
+
+
+     var a: atomic int;
+
+     coforall loc in Locales do on loc do
+       coforall 1..numTasksPerLocale do
+         for i in 1..iters do
+           a.addBuff(i);                   // buffered atomic add
+
+     flushAtomicBuff();                    // flush any pending operations (required)
+
+
+     const itersSum = iters*(iters+1)/2,   // sum from 1..iters
+           numTasks = numLocales * numTasksPerLocale;
+     assert(a.read() == numTasks * itersSum);
+
+   It's important to be aware that buffered atomic operations are not
```

Comment:
This could be read as saying that the ops don't happen until the explicit flush is done.  I'd rather not leave that misinterpretation open.  How about "may not be consistent" instead of "are not"?

Reply:
Changed to:

"""
It's important to be aware that buffered atomic operations are not consistent with regular atomic operations and updates may not be visible until the buffers are explicitly flushed
"""

I like the phrasing that buffered atomics aren't consistent with non-buffered atomics, but I agree it's important to note that buffers will be flushed automatically when full. I'm going to merge now, but I'm open to other phrasing (and am planning on having people not familiar with the implementation take a look at the docs too.)